### PR TITLE
[Attribute] Fix class name in attribute factory pass

### PR DESCRIPTION
**Pull request by Arminek**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | https://github.com/Sylius/Sylius/issues/3870 |
| License | MIT |

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
:+1:

File: src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php [hunk 1]
```diff
@@ -29,14 +32,32 @@ public function process(ContainerBuilder $container)
...
+            Factory::class,
+            array(
+                new Parameter('sylius.model.product_attribute.class'),
+            )
+        );
+
+        $translatableFactoryDefinition = new Definition(
+            TranslatableFactory::class,
+            array(
+                $baseFactoryDefinition,
```
Comment (stloyd) [inline src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php]: Shouldn't this be `Reference` not direct `Definition` injection?

Comment (Arminek) [inline src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php]: :+1:

File: src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php [hunk 2]
```diff
@@ -30,13 +34,29 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        $registry = $container->getDefinition('sylius.registry.attribute_type');
+        $baseFactoryDefinition = new Definition(
+            Factory::class,
+            array(
+                new Parameter('sylius.model.product_attribute.class'),
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php]: Attribute bundle should not know about ``product_attribute``, we have a bad coupling bug in the whole AttributeBundle...

File: src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php [hunk 3]
```diff
@@ -34,7 +34,7 @@ public function process(ContainerBuilder $container)
 
         foreach ($container->getParameter('sylius.attribute.subjects') as $subject => $configuration) {
             $oldAttributeFactory = $container->getDefinition(sprintf('sylius.factory.%s_attribute', $subject));
-            $attributeFactoryDefinition = new Definition(AttributeFactory::class);
+            $attributeFactoryDefinition = new Definition($container->getParameter('sylius.factory.product_attribute.class'));
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AttributeBundle/DependencyInjection/Compiler/RegisterAttributeFactoryPass.php]: This should not be hardcoded to product `getParameter(sprintf('sylius.factory.%s_attribute.class', $subject));`

**Pull request closed without merging**